src/UI/ContentPlaceholders.elm

```diff
@@ -0,0 +1,236 @@
+module UI.ContentPlaceholders exposing
+    ( ContentPlaceholder
+    , custom
+    , nothingToSeeHere
+    , withSize
+    , renderElement
+    )
+
+{-| `UI.ContentPlaceholders` fills space in an area that is waiting for some action from the user.
+
+An empty-state can be created and rendered as in the following pipeline:
+
+    ContentPlaceholders.custom
+        { icon = Icon.fix
+        , title = "Select A Group"
+        , body = "Please select a group to fix from the list on the left."
+        }
+        |> ContentPlaceholders.withSize Size.large
+        |> ContentPlaceholders.renderElement renderConfig
+
+There are also some ready-to-use costructors.
+
+
+# Building
+
+@docs ContentPlaceholder
+@docs custom
+
+
+# Common values
+
+@docs nothingToSeeHere
+
+
+# Size
+
+@docs withSize
+
+
+# Rendering
+
+@docs renderElement
+
+-}
+
+import Element exposing (Element, fill, maximum)
+import Element.Font as Font
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.RenderConfig exposing (localeTerms)
+import UI.Internal.Size exposing (Size(..))
+import UI.Palette as Palette exposing (brightnessLight, toneGray)
+import UI.RenderConfig as RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.Utils.Element as Element
+
+
+{-| The `ContentPlaceholder` type is used for describing the component for later rendering.
+-}
+type ContentPlaceholder
+    = Custom Properties Options
+    | NothingToSeeHere Options
+
+
+type alias Properties =
+    { icon : String -> Icon
+    , title : String
+    , body : String
+    }
+
+
+type alias Options =
+    { size : Size }
+
+
+{-| Constructs a generic `ContentPlaceholder`.
+
+    ContentPlaceholders.custom
+        { icon = Icon.download
+        , title = "Pick a file"
+        , body = "Please select a file to download it."
+        }
+        |> ContentPlaceholders.renderElement renderConfig
+
+-}
+custom :
+    { icon : String -> Icon
+    , title : String
+    , body : String
+    }
+    -> ContentPlaceholder
+custom prop =
+    Custom prop defaultOptions
```

Comment:
is it meant to support a custom placeholder? Will this be used anywhere?

Reply:
Yep, here: https://www.figma.com/file/DTyoSveNksHpZCdU8a7BNs/Dashboard?node-id=1855%3A26417